unity-sdk/AndroidPlugin/OptimoveUnityPlugin/src/main/java/com/optimove/unity/plugin/UnityProxy.java

```diff
@@ -1,4 +1,376 @@
+            Optimove.getInstance().setUserId(userId);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    public static void setUserEmail(String userEmail) {
+        try {
+            Optimove.getInstance().setUserEmail(userEmail);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    public static void reportEvent(String eventName) {
+        try {
+            Optimove.getInstance().reportEvent(eventName);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    public static void reportEvent(String eventName, String eventParamsString) {
+
+        if (eventParamsString == null) {
+            reportEvent(eventName);
+            return;
+        }
+
+        try {
+            JSONObject eventParamsJson = new JSONObject(eventParamsString);
+            Iterator<String> iterator = eventParamsJson.keys();
+            Map<String, Object> eventParams = new HashMap<>();
+            while (iterator.hasNext()) {
+                String key = iterator.next();
+                eventParams.put(key, eventParamsJson.get(key));
+            }
+            Optimove.getInstance().reportEvent(eventName, eventParams);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    public static void registerUser(String userId, String userEmail) {
+        try {
+            Optimove.getInstance().registerUser(userId, userEmail);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+
+    }
+
+    public static void reportScreenVisit(String screenName) {
+        try {
+            Optimove.getInstance().reportScreenVisit(screenName);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+
+    }
+
+    public static void reportScreenVisit(String screenName, String screenCategory) {
+        try {
+            Optimove.getInstance().reportScreenVisit(screenName, screenCategory);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    public static String getVisitorId() {
+        try {
+
+            return Optimove.getInstance().getVisitorId();
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+        return "";
+    }
+
+    public static void pushRegister() {
+        try {
+            Optimove.getInstance().pushRegister();
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    public static void pushUnregister() {
+        try {
+
+            Optimove.getInstance().pushUnregister();
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+
+    }
+
+    public static void inAppUpdateConsent(boolean consented) {
+        try {
+
+            OptimoveInApp.getInstance().updateConsentForUser(consented);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+
+    }
+
+    public static String getCurrentUserId() {
```

Comment:
is this part of the API ? in cordova `getCurrentUserIdentifier` is only present on android and i bet was just erroneously not removed, can double check?

Reply:
I think you are right, it's not in our public API functions list, I removed it